plugins/webhooks/index.js

```diff
@@ -323,47 +412,50 @@ function pushEvent(pluginOptions, request, reply, parsed) {
     const eventFactory = request.server.app.eventFactory;
     const pipelineFactory = request.server.app.pipelineFactory;
     const userFactory = request.server.app.userFactory;
-    const { hookId, checkoutUrl, branch, sha, username, scmContext, changedFiles } = parsed;
+    const { hookId, checkoutUrl, branch, username, scmContext } = parsed;
     const fullCheckoutUrl = `${checkoutUrl}#${branch}`;
+    const scmConfig = {
+        scmUri: '',
+        token: ''
+    };
 
     request.log(['webhook', hookId], `Push for ${fullCheckoutUrl}`);
 
     // Fetch the pipeline associated with this hook
     return obtainScmToken(pluginOptions, userFactory, username, scmContext)
-        .then(token => pipelineFactory.scm.parseUrl({
-            checkoutUrl: fullCheckoutUrl,
-            token,
-            scmContext
-        }))
-        .then(scmUri => pipelineFactory.get({ scmUri }))
-        .then((pipeline) => {
-            if (!pipeline) {
+        .then((token) => {
+            scmConfig.token = token;
+
+            return pipelineFactory.scm.parseUrl({
+                checkoutUrl: fullCheckoutUrl,
+                token,
+                scmContext
+            });
+        }).then((scmUri) => {
+            scmConfig.scmUri = scmUri;
+
+            return triggerPipelines(pipelineFactory, scmConfig, branch);
+        }).then((pipelines) => {
+            if (!pipelines || pipelines.length <= 0) {
                 request.log(['webhook', hookId],
                     `Skipping since Pipeline ${fullCheckoutUrl} does not exist`);
 
+                return [];
+            }
+
+            return createEvents(eventFactory, pipelines, parsed);
+        })
+        .then((events) => {
+            if (events.length <= 0) {
```

Comment:
Is there a case that `events.length` is minus?

Reply:
The length will never be minus, it's kind my habit.  Is it better to compare by `events.length === 0`?

Reply:
I think so.

Reply:
Ok then I will fix.